ambry-api/src/main/java/com.github.ambry/router/CryptoService.java

```diff
@@ -30,11 +32,27 @@
    * @param toEncrypt {@link ByteBuffer} that needs to be encrypted
    * @param key the secret key (of type T) to use to encrypt
    * @return the {@link ByteBuffer} containing the encrypted content. Ensure the result has all
-   * the information like the IV along with the encrypted content, inorder to decrypt the content with a given key
+   * the information like the IV along with the encrypted content, in order to decrypt the content with a given key
    * @throws {@link GeneralSecurityException} on any exception with encryption
    */
   ByteBuffer encrypt(ByteBuffer toEncrypt, T key) throws GeneralSecurityException;
 
+  /**
+   * Encrypts the {@code toEncrypt} with the given key. The {@code toEncrypt} is a netty {@link ByteBuf}. This is used
+   * for encrypting data chunks.
+   * @param toEncrypt {@link ByteBuf} that needs to be encrypted
+   * @param key the secret key (of type T) to use to encrypt
+   * @return the {@link ByteBuf} containing the encrypted content. Ensure the result has all
+   * the information lik the IV along with the encrypted content, in order to decrypt the content with a given key
+   * @throws {@link GeneralSecurityException} on any exception with encryption
+   */
+  default ByteBuf encrypt(ByteBuf toEncrypt, T key) throws GeneralSecurityException {
+    ByteBuffer encrypted = encrypt(toEncrypt.nioBuffer(), key);
+    ByteBuf result = ByteBufAllocator.DEFAULT.ioBuffer(encrypted.remaining());
```

Comment:
Why not use `Unpooled.wrappedBuffer` in this case instead of copying?

Reply:
We should use the Unpooled ByteBuf since it's just default method which will be override by the actually implementation.